examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml

```diff
@@ -26,4 +26,3 @@ spec:
   tableIndex:
     hashKeyName: hashKey
     hashKeyType: S
-  
```

Comment:
Why are newlines at the end being removed? We should keep them.

Reply:
Its deleting extra lines which are more than one in each file. I can see the empty line in the code but not sure why its displaying like this in the PR.

We are using  `pre-commit-hooks` with `end-of-file-fixer` to remove unnecessary extra lines and keeps  only one new line

Reply:
It deleted the last newlines in files that were supposed to be there. See https://raw.githubusercontent.com/aws-samples/crossplane-aws-blueprints/main/examples/aws-provider/composite-resources/dynamodb/dynamodb-provisioned-composite-gsi.yaml